### ISSUE TITLE
*.field_to_match.headers.match_pattern "all" option is not working

### ISSUE THREAD

**Issue by felipe88alves** [umotif-public/terraform-aws-waf-webaclv2#94]
**What is the current behavior?**
Applying a rule with the `field_to_match` set to `headers` will fail when the `match_pattern` is set to `all`

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
The following rule will not fail a `plan`, but will produce an empty statement, which fails when running an `apply`
```
    {
      name     = "custom_block_rule_1"
      priority = 1
      action   = "block"
      statement = {
        byte_match_statement = {
          field_to_match = {
            headers = {
              match_pattern     = {
                "all" = {}
              }
              match_scope       = "ALL"
              oversize_handling = "CONTINUE"
            }
          }
          positional_constraint = "CONTAINS"
          search_string         = "string"
          priority              = 0
          type                  = "NONE"
        }
      }
      visibility_config = {
        sampled_requests_enabled   = true
        cloudwatch_metrics_enabled = true
        metric_name                = "custom_block_rule_1"
      }
    }
```

**What is the expected behavior?**
It should not produce an empty statement and should not fail an `apply`

**Software versions?**
umotif/terraform-aws-waf-webaclv2: v4.2.0
hashicorp/aws: v4.59.0

**Comment by MatanHeledPort** [comment]
Hey, I created a PR which I think is supposed to fix this issue?
https://github.com/umotif-public/terraform-aws-waf-webaclv2/pull/95

**Comment by felipe88alves** [comment]
Thanks @MatanHeledPort!
Did you test the solution?

I was thinking of something similar to what's done with the cookies:
`for_each = contains(keys(match_pattern.value), "all") ? [lookup(match_pattern.value, "all")] : []`

**Comment by MatanHeledPort** [comment]
Maybe keeping convention is better - I will change it to match `cookies`

**Comment by MatanHeledPort** [comment]
For some reason it doesnt behave the same as cookies
![image](https://user-images.githubusercontent.com/115919235/233102607-f43d8059-b5d6-4292-9d3e-9535bcfed2a5.png)

**Comment by felipe88alves** [comment]
seems like the problem is with your use of the statement block.
Want to share you config here?

**Comment by felipe88alves** [comment]
Would probably be good to add an example as well in https://github.com/umotif-public/terraform-aws-waf-webaclv2/blob/main/examples/wafv2-bytematch-rules/main.tf

**Comment by MatanHeledPort** [comment]
Sure!
This is the scope_down_statement:
```
        scope_down_statement = {
          and_statement = {
            statements = [
              {
                not_statement = {
                  byte_match_statement = {
                    field_to_match = {
                      headers = {
                        match_pattern = {
                          all = {}
                        }
                        match_scope = "ALL"
                        oversize_handling = "CONTINUE"
                      }
                    }
                  positional_constraint = "CONTAINS"
                  search_string         = "authorization"
                  priority              = 0
                  type                  = "NONE"
                  }
                }
              },
              {
                not_statement = {
                  byte_match_statement = {
                    field_to_match = {
                      uri_path = "{}"
                    }
                  positional_constraint = "EXACTLY"
                  search_string         = "/"
                  priority              = 0
                  type                  = "NONE"
                  }
                }
              }
            ]
          }
        }
```

Working on an example aswell

**Comment by MatanHeledPort** [comment]
Yes, exactly

**Comment by felipe88alves** [comment]
Took a quick look here, seems like the problem is with your use of `override_action`.
It is described as a dynamic field in the `main.tf` schema, whereas you use it as a normal var in the example.

Replace `override_action = "count"` to `action = "count"` and it should work

**Comment by MatanHeledPort** [comment]
Ok that seemed to do the trick! 
The change worked

**Comment by felipe88alves** [comment]
I'm don't have write access, but maybe @Ohid25 can help 👍

**Comment by felipe88alves** [comment]
Can we please get some feedback on this PR?
Really needing it to reduce some manual toil.
Cheers

**Comment by MatanHeledPort** [comment]
Feel free to use my forked PR branch. Its what Im doing untill this merges @felipe88alves

**Comment by Ohid25** [comment]
This should now be fixed as a part of [4.4.0](https://github.com/umotif-public/terraform-aws-waf-webaclv2/releases/tag/4.4.0).
